sdk/servicebus/Azure.Messaging.ServiceBus/tests/Infrastructure/ServiceBusScope.cs

```diff
@@ -0,0 +1,627 @@
+                              bool shouldRemoveAtScopeCompletion)
+            {
+                NamespaceName = serviceBusNamespaceName;
+                TopicName = topicName;
+                SubscriptionNames = subscriptionNames;
+                ShouldRemoveAtScopeCompletion = shouldRemoveAtScopeCompletion;
+            }
+
+            /// <summary>
+            ///   Performs the tasks needed to remove the dynamically created Service Bus Queue.
+            /// </summary>
+            ///
+            public async ValueTask DisposeAsync()
+            {
+                if (!ShouldRemoveAtScopeCompletion)
+                {
+                    _disposed = true;
+                }
+
+                if (_disposed)
+                {
+                    return;
+                }
+
+                try
+                {
+                    var subscription = TestEnvironment.ServiceBusSubscription;
+                    var resourceGroup = TestEnvironment.ServiceBusResourceGroup;
+                    var token = await AquireManagementTokenAsync().ConfigureAwait(false);
+
+                    using var client = new ServiceBusManagementClient(new TokenCredentials(token)) { SubscriptionId = subscription };
+                    await CreateRetryPolicy().ExecuteAsync(() => client.Topics.DeleteAsync(resourceGroup, NamespaceName, TopicName)).ConfigureAwait(false);
+                }
+                catch
+                {
+                    // This should not be considered a critical failure that results in a test failure.  Due
+                    // to ARM being temperamental, some management operations may be rejected.  Throwing here
+                    // does not help to ensure resource cleanup only flags the test itself as a failure.
+                    //
+                    // If a queue fails to be deleted, removing of the associated namespace at the end of the
+                    // test run will also remove the orphan.
+                }
+
+                _disposed = true;
+            }
+        }
+
+        /// <summary>
+        ///   An internal type for tracking the management access token and
+        ///   its associated expiration.
+        /// </summary>
+        ///
+        private class ManagementToken
+        {
+            /// <summary>The value bearer token to use for authorization.</summary>
+            public readonly string Token;
+
+            /// <summary>The date and time, in UTC, that the token expires.</summary>
+            public readonly DateTimeOffset ExpiresOn;
+
+            /// <summary>
+            ///   Initializes a new instance of the <see cref="ManagementToken"/> class.
+            /// </summary>
+            ///
+            /// <param name="token">The value of the bearer token.</param>
+            /// <param name="expiresOn">The date and time, in UTC, that the token expires on.</param>
+            ///
+            public ManagementToken(string token,
```

Comment:
I see you are holding on to the EH way of formatting parameters 😆

Reply:
or was EH using the me approach to formatting parameters?   Inception.  👀